lib/bolt/plugin/terraform.rb

```diff
@@ -49,11 +71,41 @@ def inventory_targets(opts)
       end
 
       def load_statefile(opts)
+        statefile = if opts['backend'] == 'remote'
+                      load_remote_statefile(opts)
+                    else
+                      load_local_statefile(opts)
+                    end
+
+        JSON.parse(statefile)
+      end
+
+      # Uses the Terraform CLI to pull remote state files
+      def load_remote_statefile(opts)
+        dir = File.expand_path(opts['dir'])
+
+        begin
+          stdout_str, _stderr_str, status = Open3.capture3('terraform state pull', chdir: dir)
+        rescue StandardError => e
+          raise Bolt::Error.new(
+            "Failed to load remote state file: #{e.message}",
+            'FILE_ERROR'
+          )
+        end
+
+        unless status.success?
+          err = stdout_str.split("\n").first
```

Comment:
Depending on any errors that terraform raises, it might print to one or both of `stdout` and `stderr`. Since these errors can be raised for a couple reasons (terraform hasn't been initialized, missing files, etc.) would it make sense to display all of the text that terraform prints to `stdout` and `stderr`, or just have a generic message saying there was an issue with the terraform configuration and to sort that out separately?

Reply:
You could grep the dir name in the error message. But i guess if the dirname was terraform that might be confusing. In the rescue block you could just do `File.directory?(dir)` and change the message based on that.

Reply:
I guess that only addresses the first part of the question. I think surfacing the error message terraform displays is important. We just want to be as helpful as possible.